FEAT-#4524: Split Modin API and Memory log files (#4526)
modin-project/modin: docs/usage_guide/advanced_usage/modin_logging.rst

Modin Logging
=============

Modin logging offers users greater insight into their queries by logging internal Modin API calls, partition metadata,
and profiling system memory. When Modin logging is enabled (default disabled), log files are written to a local ``.modin`` directory at the same
directory level as the notebook/script used to run Modin. It is possible to configure whether to log system memory and additional metadata
in addition to Modin API calls (see the usage examples below).

The logs generated by Modin Logging will be written to a ``.modin/logs/job_<uuid>`` directory, uniquely named after the job uuid.
The logs that contain the Modin API stack traces are named ``trace.log``. The logs that contain the memory utilization metrics are
named ``memory.log``. By default, if any log file exceeds 10MB, the logs will rollover and several files will be created. For instance, if
users have 20MB worth of Modin API logs, they can expect to find ``trace.log.1`` and ``trace.log.2`` in the ``.modin/logs/job_<uuid>`` directory.

**Developer Warning:** In some cases, running services like JupyterLab in the ``modin/modin`` directory may result in circular dependency issues.

> This is as a result of a name conflict between the ``modin/logging`` directory and the Python ``logging`` module, which may be used as a default in

Reviewer: ```suggestion
This is due to a naming conflict between the ``modin/logging`` directory and the Python ``logging`` module, which may be used as a default in
```

Modin logging offers users greater insight into their queries by logging internal Modin API calls, partition metadata, 
and profiling system memory. When Modin logging is enabled (default disabled), log files are written to a local `.modin` directory at the same
directory level as the notebook/script used to run Modin. It is possible to configure whether to log system memory and additional metadata 
in addition to Modin API calls (see the usage examples below).

The logs generated by Modin Logging will be written to a `.modin/logs/job_<uuid>` directory, uniquely named after the job uuid.
The logs that contain the Modin API stack traces are named `trace.log`. The logs that contain the memory utilization metrics are 
named `memory.log`. By default, if any log file exceeds 10MB, the logs will rollover and several files will be created. For instance, if 

> users have 20MB worth of Modin API logs, you can expect to find `trace.log.1` and `trace.log.2` in the `.modin/logs/job_<uuid>` directory.

Reviewer: ```suggestion
users have 20MB worth of Modin API logs, they can expect to find `trace.log.1` and `trace.log.2` in the `.modin/logs/job_<uuid>` directory.
```

**Developer Warning:** In some cases, running services like JupyterLab in the `modin/modin` directory may result in circular dependency issues. 
This is as a result of a name conflict between the `modin/logging` directory and the Python `logging` module, which may be used as a default in 

> such environments. As a resolution, one could run Jupyterlab or other services from any directory except `modin/modin`. 

Reviewer: ```suggestion
such environments. To resolve this, please run Jupyterlab or other similar services from directories other than  `modin/modin`. 
```

> such environments. As a resolution, one one could run Jupyterlab or other services from any directory except `modin/modin`. 

Reviewer 2: ```suggestion
such environments. As a resolution, one could run Jupyterlab or other services from any directory except `modin/modin`. 
```
